File: DesktopCore/src/bean/ComboBoxNumberComparison.java
package bean;

import java.awt.Dimension;

import javax.swing.DefaultComboBoxModel;
import javax.swing.JComboBox;

public class ComboBoxNumberComparison extends JComboBox<String>
{
	public static final String FILTER_MODE_EQUAL_TO = "=";
	public static final String FILTER_MODE_GREATER_THAN = ">";
	public static final String FILTER_MODE_LESS_THAN = "<";
	
	private final String[] filterModes = {FILTER_MODE_EQUAL_TO, FILTER_MODE_GREATER_THAN, FILTER_MODE_LESS_THAN};
	
	public ComboBoxNumberComparison()
	{
		super();
		setModel(comboModel);
		setPreferredSize(new Dimension(40, 30));
		setMaximumSize(new Dimension(40, 30));
	}
	
	private final DefaultComboBoxModel<String> comboModel = new DefaultComboBoxModel<>(filterModes);
	
	public void addOption(String option)
	{
		comboModel.addElement(option);
	}
}

File: Darts/src/main/kotlin/dartzee/core/bean/ComboBoxNumberComparison.kt
package dartzee.core.bean

import java.awt.Dimension

import javax.swing.DefaultComboBoxModel
import javax.swing.JComboBox

class ComboBoxNumberComparison : JComboBox<String>()
{
    private val filterModes = arrayOf(FILTER_MODE_EQUAL_TO, FILTER_MODE_GREATER_THAN, FILTER_MODE_LESS_THAN)
    private val comboModel = DefaultComboBoxModel(filterModes)

    init
    {
        model = comboModel
        preferredSize = Dimension(40, 30)
        maximumSize = Dimension(40, 30)
    }

    fun addOption(option: String)
    {
        comboModel.addElement(option)
    }

    companion object
    {
        const val FILTER_MODE_EQUAL_TO = "="
        const val FILTER_MODE_GREATER_THAN = ">"
        const val FILTER_MODE_LESS_THAN = "<"
    }
}
